server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -1262,11 +1263,13 @@ def todate(
                     convert_fn = StrToDate
                 case ToDateOp.PARSE_DATE:
                     convert_fn = ParseDate
+                case ToDateOp.TO_TIMESTAMP:
+                    convert_fn = functions.Timestamp
                 case _:
                     raise NotImplementedError(f"[{self.DIALECT}] todate has no set operator")
             date_selection = convert_fn(col_field, step.format)
         else:
```

Comment:
We're still returning something that's not a timestamp if the format is specified. Could you please add `self._cast_to_timestamp` of the result in both cases ?